### DISCUSSION TITLE
v1.0.0

### DISCUSSION THREAD

**Discussion by yupix** [yupix/Mi.py#65] [Announcements]
### Added
- `Note` クラスに `reply` メソッドを追加しました
- 以下のクラスを追加
    - MisskeyWebSocket
    - MisskeyClientWebSocketResponse
    - Route
    - HTTPClient
- Clientに以下のメソッドを追加
    - post_chat
    - delete_chat
    - post_note
        - 今後のノート投稿はこちらを使用してください 
    - delete_note
    - get_instance
    - fetch_instance
    - get_user
    - fetch_user
    - upload_file
- `rich` を使ったデバッグ機能を追加しました
- `file_upload` 関数を追加しました
- `run` メソッドに `debug` 引数を追加しました
- DocStringを一部追加しました
- 抽象基底クラスを追加しました
- TypedDictを追加しました
- チャットをする際に用いる `Chat` `ChatContent` クラスを追加しました
- `on_chat` イベントを追加しました
- ~~`NoteContent` クラスを追加しました~~
    ~~- ノートの受信イベントでは基本このクラスが使用されます。~~
- `Instance` クラスに `get_users` メソッドを追加しました

### Changed

- api周りでのエラー出力が分かりやすくなりました
- ~~今までの `Note` クラスに当たるものを `NoteContent` に変更し別途 `Note` クラスを作成させるようにしました~~
    - ~~これは不要なデータなどを生成しないようにすることが目的です。~~
- 部分的にPydanticを廃止
    - 素直に使わないほうが部分的に楽だから
- `Drive` クラスの `upload` メソッドで例外 `InvalidParameters` を発生させるようにしました
    - これは`to_file`, `to_url` の両方がないと変数が定義されず、`Drive`クラスの生成に失敗する可能性があるからです。
- `get_user` メソッドが 非同期になりました。
- 内部変更: `utils.py` の一部を Cython を用いたものに変更
- 内部変更: `DriveAction` を廃止し, `file_upload` を使用するように
- 内部変更: `event_dispatch` でクラスにイベントがある場合呼び出すようにしました
- 内部変更: シングルクォーテーションをダブルクォーテーションに変更しました
- 内部変更: cogシステムを作り直しました
- 内部変更: websocketに関連する部分を作り直しました
- 内部変更: イベントの発火部分を作り直しました
- 内部的変更: `dispatch` に関連する物を非同期から同期に変更しました
- **破壊的変更**: 使用しているライブラリを `websockets` から `aiohttp` に変更しました
- **破壊的変更**: `Router` クラスの `channels` メソッドを `connect_channel` に変更しました
- **破壊的変更**: taskをtasksに変更しました
- **破壊的変更**: `on_ready` を除きwebsocketを引数で渡さないようになりました
- **破壊的変更**: `Note` クラスの `text` 変数を `content` と統合しました
- **破壊的変更**: `Reaction` クラスと `ReactionContent` を結合させました
- **破壊的変更**: `Chat` クラスと `ChatContent` を結合させました
- **破壊的変更**: `Drive` クラスの `upload` メソッドで使用できる引数名を変更、キーワード引数を強制するようにしました。
    - 引数名の変更は次の通りです `url` => `to_url`, `path` => `to_file`
    - 強制されるキーワード引数は次の通りです `force`, `is_sensitive`

### Fixed

- __on_error が動かない不具合を修正しました
- 一部の誤った typing hintを修正
- バグの原因になるコードを修正

### Removed

- `Router` クラスから以下のメソッドを削除しました
    - main_channel
    - home_time_line
    - local_time_line
    - global_time_line
- クラスの削除について
    - `ChatContent`
    - `ReactionContent`
    - `NoteContent`
    - `UserAction`
    - `UserProfile`
        - `UserProfile`にあったものは `User` に統合され、詳細な情報は `User` クラスの detailsオブジェクト(`UserDetails`) から取得可能です
- **破壊的変更**: `WebSocket` クラスを削除しました
    - 今後はaiohttpの `ws_connect` メソッドを使用します
- **破壊的変更**: `requests` ライブラリを削除しました
    - 今後はaiohttpの `request` メソッドを使用します

<hr /><em>This discussion was created from the release <a href='https://github.com/yupix/Mi.py/releases/tag/v1.0.0'>v1.0.0</a>.</em>

**Comment by yupix** [comment]
This release is currently not available for upload due to the author's environment and does not exist on pypi.

**Reply by yupix** [reply]
Uploaded now that the problem has been resolved.

**Comment by fotoente** [comment]
As the `Router` function got removed, the example throws an error on startup.
Seems like it can't post a note.

Loop works great!
And it reacts to new emojis!

**Reply by yupix** [reply]
Can you show me where I can find an example of using a router?
